### DISCUSSION TITLE
Field validation

### DISCUSSION THREAD

**Discussion by GrahameArmitage** [Blazored/FluentValidation#176] [Q&A]
Hello there.

Please can someone help me out?

I have a form with a button that performs validation and it works fine. I'm after a way of intercepting validation when a field is exited so I can return focus back to the field if it's invalid. The default way is the OnFieldChanged() event is fired, does the validation (including a call to the api for uniqueness), highlights the field red if invalid and goes to the next field.

Any help would be most appreciated.

Cheers,

Grahame

**Comment by Sharaf-Mansour** [comment]
I think it will be custom built functionality I will try something out and give you feedback.

**Comment by Sharaf-Mansour** [comment]
@GrahameArmitage Update.
After 2 hours here is what I got to set auto focus on the 1st invalid input after submit.
But it is very messy.

## Assumptions
I assume that you know how to expose your validations to the Razor Code 
```cs
bool isValid(object SomeObj) => SomeObjValidator.Validate(SomeObj).IsValid;
```
and I assume you know how to use the ```@ref``` and [ElmentRefrance ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.elementreference?view=aspnetcore-7.0)in Razor. 

## Now to the mess

I created a custom class;
```cs
class FocusInput
  {
      public string Name { get; set; }
      public dynamic Element { get; set; }
      public  ValueTaskAwaiter Focus() => 
      Element switch
       {
          InputText => ((InputText)Element).Element.Value.FocusAsync().GetAwaiter(),
          InputDate<DateTime> => ((InputDate<DateTime>)Element).Element.Value.FocusAsync().GetAwaiter(),
          InputSelect<string> => ((InputSelect<string>)Element).Element.Value.FocusAsync().GetAwaiter(),
          _ => ((ElementReference)Element).FocusAsync().GetAwaiter(),
      };
  }
  ```
  Then I created a List<FocusInput> to work as a container for all my inputs. 
  and bind the Element using **@ref** as for the Name, it is the name of the property to be validated assuming it is unique e.g. (Password, Email, etc...) 
```cs 
nameof(model.Username)
```
  this is just the easy part.
The real pain (For me) is to set a custom validator for every single input. so I can track its status via its name.
Something like this
```cs
bool isValid(object SomeObj) => SomeObjValidator.Validate(SomeObj).IsValid;

if (!IsValid(model.Password))
focusInputs.FirstOrDefault(x => x.Name = nameof(model.Password)).Focus();
```
This is not the cleanest way to do it nor the best.
